CMakeLists.txt

```diff
@@ -111,7 +87,7 @@ endif()
 message(STATUS "##################################################")
 message(STATUS "Mumble version:  ${version}")
 message(STATUS "Architecture:    ${ARCH}")
-message(STATUS "Build type:      ${BUILD_STR}")
+message(STATUS "Build type:      ${CMAKE_BUILD_TYPE}")
```

Comment:
If no type was specified, I think this will just result in an empty string which in turn seems like our mechanism for reporting the build type is broken. That's why I wasn't using this variable directly before.

Reply:
I guess we need to discover what specifics we have defined that depend on the build type. Setting this during the generation step is only recognized by single config generators. Multi config generators add 4 build types that need to be selected at the build step. Since the hard define is not working with NMake Makefiles unless the build type is specified, we may not be getting a consistent result already.

Reply:
Yeah that's why I had all these different cases to report the build type which included the multi-generator one :eyes:

Reply:
The hard define for `CMAKE_BUILD_TYPE=Release` has been added back as suggested. I think that should be a valid compromise for this situation. Since, `Debug`, `null`, or `Release` are all valid that should be sufficient. We cover the two build types we support in the vcpkg document and that should probably be highlighted for CMake overall.

Reply:
I think for printing out the build type we should take into account qhether the used generator is a multi config one.in which case we shouldn't claim to be building a specific config :thinking:

Reply:
I think solving it like this would probably cover all cases:
```
get_property(IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)

message(STATUS "##################################################")
message(STATUS "Mumble version:  ${version}")
message(STATUS "Architecture:    ${ARCH}")
if(NOT IS_MULTI_CONFIG)
    message(STATUS "Build type:      ${CMAKE_BUILD_TYPE}")
else()
    message(STATUS "Using multi-config generator that will determine build type on-the-fly")
endif()
```

Reply:
Added.